actions/inventory.go

```diff
@@ -245,15 +246,14 @@ func (a *InventoryCollectorAction) Collect(ctx context.Context, device *common.D
 	// Update DriveCollectors based on drive vendor attributes
 	if a.dynamicCollection {
 		for _, sc := range a.device.StorageControllers {
-			if sc.SupportedRAIDTypes == "" {
-				continue
-			}
+			c := DriveCollectorByStorageControllerVendor(sc.Vendor, a.trace)
 
-			a.collectors.DriveCollectors = append(
-				a.collectors.DriveCollectors,
-				DriveCollectorByStorageControllerVendor(sc.Vendor, a.trace),
-			)
+			if c != nil {
+				a.collectors.DriveCollectors = append(a.collectors.DriveCollectors, c)
+			}
+		}
 
+		if len(a.collectors.DriveCollectors) > 0 {
```

Comment:
return early so we avoid larger nested blocks of conditionals

```suggestion
		if len(a.collectors.DriveCollectors) == 0 {
            continue
        }
```

Reply:
Since this code isn't within a for loop, and there's additional stuff further on in the function I wouldn't want to skip, I think this code is best as-is.